pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -77,10 +76,7 @@ func getPreFilterState(cycleState *framework.CycleState) (preFilterState, error)
 
 	c, err := cycleState.Read(preFilterStateKey)
 	if err != nil {
-		// The metadata wasn't pre-computed in prefilter. We ignore the error for now since
-		// Filter is able to handle that by computing it again.
-		klog.V(5).Infof("Error reading %q from cycleState: %v", preFilterStateKey, err)
-		return nil, nil
+		return nil, fmt.Errorf("Error reading %q key from cycleState: %v", preFilterStateKey, err)
```

Comment:
Q: are we going to follow this pattern (return err when not finding the state entry) in all plugins?

I ask that is because there are some places still tolerating the absent entry.

Reply:
no, this is wrong. I reverted the change and added unit tests. I also added unit tests to interpod affinity to make sure we catch that as well.